### DISCUSSION TITLE
flat_hash_map string type serialization

### DISCUSSION THREAD

**Discussion by mjshare** [abseil/abseil-cpp#1489] [Q&A]
I plan to serialize flat_hash_map and store it. It can be restored by load in the future. Currently, a problem is found. When the type is a string, the serialized data does not contain string data. As a result, the serialized data is lost. How do we serialize the string type?
and which interface is used to deserialize to flat_hash_map

**Comment by derekmauro** [comment]
Serialization of our hash containers is intentionally not supported. Both `absl::Hash` and the hash containers themselves inject instability into the hash ordering to prevent users from depending on their implementation.

See https://abseil.io/resources/swe-book/html/ch01.html#example_hash_ordering for more information.

You can manually store the key-value pairs using the standard API.
